api/control/v1alpha1/types.go

```diff
@@ -302,6 +315,211 @@ type BindingList struct {
 	Items           []Binding `json:"items"`
 }
 
+// StatusCollector defines one way to collect status about a given workload object from
+// the set of WECs that it propagates to.
+// This is modeled after an SQL SELECT statement that does aggregation.
+//
+// +kubebuilder:object:root=true
+type StatusCollector struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   StatusCollectorSpec   `json:"spec,omitempty"`
+	Status StatusCollectorStatus `json:"status,omitempty"`
+}
+
+// StatusCollectorSpec defines the desired state of StatusCollector.
+type StatusCollectorSpec struct {
+	// `filter`, if given, is applied first.
+	// It must evaluate to a boolean or null (which is treated as false).
+	// This is like the WHERE clause in an SQL SELECT statement.
+	// +optional
+	Filter *Expression `json:"filter,omitempty"`
+
+	// `groupBy` says how to group workload objects for aggregation (if there is any).
+	// Each expression must evaluate to an atomic value.
```

Comment:
I would add a comment that `groupBy` must be empty when `combinedFields` is.

Reply:
done.

Reply:
Looks like you read a "not" that I did not write.

Reply:
fixed.